torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -286,11 +320,38 @@ class AttributePropagator {
               paramConst->debugName());
           n->outputs().at(0)->replaceAllUsesWith(paramConst);
           n->removeAllInputs();
+        } else if (n->kind() == prim::fork) {
+          auto func = [this](std::shared_ptr<Graph>& subgraph) {
+            propagateAttributes(subgraph);
+          };
+          applyToSubGraph(n, graph, func);
         }
       }
     }
   }
 
+  void applyToSubGraph(
+      Node* n,
+      std::shared_ptr<Graph>& graph,
+      const std::function<void(std::shared_ptr<Graph>&)>& func) {
+    TORCH_CHECK(n->kind() == prim::fork);
+    auto attrModule = module_;
+    auto node = n->inputs()[0]->node();
+    if (node->kind() != prim::GetAttr)
+      return;
+    auto name = node->s(attr::name);
+    auto input = node->inputs()[0];
+    if (!findConstantAttr(input, name, attrModule, graph)) {
```

Comment:
I think even if the module is a constant attribute we still need to run `recordMutableAttrs` on the subgraph here, so that modified attributes/submodules within the subgraph arrer preserved.